sw/otbn/code-snippets/barrett384.s

```diff
@@ -231,23 +231,52 @@ wrap_barrett384:
 
   /* load first operand from dmem to [w9, w8] */
   li x4, 8
-  bn.lid x4++, 0(x0)
-  bn.lid x4++, 32(x0)
+  la x5, inp_a
+  bn.lid x4++, 0(x5)
+  bn.lid x4++, 32(x5)
   /* load second operand from dmem to [w11, w10] */
-  bn.lid x4++, 64(x0)
-  bn.lid x4++, 96(x0)
+  la x5, inp_b
+  bn.lid x4++, 0(x5)
+  bn.lid x4++, 32(x5)
   /* load modulus from dmem to [w13, w12] */
-  bn.lid x4++, 256(x0)
-  bn.lid x4++, 288(x0)
+  la x5, inp_m
+  bn.lid x4++, 0(x5)
+  bn.lid x4++, 32(x5)
   /* load barrett precomputed parameter u from dmem to [w15, w14] */
-  bn.lid x4++, 320(x0)
-  bn.lid x4++, 352(x0)
+  la x5, inp_u
+  bn.lid x4++, 0(x5)
+  bn.lid x4++, 32(x5)
 
   jal x1, barrett384
 
   /* store result from [w30, w29] to dmem */
   li x4, 29
-  bn.sid x4++, 512(x0)
-  bn.sid x4++, 544(x0)
+  la x5, oup_c
+  bn.sid x4++, 0(x5)
+  bn.sid x4++, 32(x5)
 
   ecall
+
+.data
+
+/* Input and output operands */
+.globl inp_a
```

Comment:
nit: would be nice to replace this comment with comments above each operand saying what they are.
```suggestion
/* First operand for Barrett modular multiplication. */
.globl inp_a
```